tests/tests.py

```diff
@@ -1,7 +1,385 @@
+        signal.signal(signal.SIGINT, signal.SIG_DFL)
+
+    end = time.monotonic_ns()
+    duration = end - start
+    print("")
+    print("*********************************")
+    print("********* TESTS SUMMARY *********")
+    print("*********************************")
+    print(
+        f"Time Spent   : {duration / 1000000000:.2f}s (cumulative: {stats.total_duration / 1000000000:.2f}s)"
+    )
+    print(f"Tests Passed : {stats.pass_tests}")
+    print(f"Tests Skipped: {stats.skip_tests}")
+    print(f"Tests Failed : {stats.fail_tests}")
+
+    if stats.fail_tests != 0:
+        print("\nFailing tests:")
+        for test_case in stats.fail_tests_names:
+            print(f"- {test_case}")
+        sys.exit(1)
+
+
+def get_gdb_host(args: argparse.Namespace, local_pwndbg_root: Path) -> TestHost:
+    """
+    Build a GDB-based test host.
+    """
+    if args.nix:
+        # Use pwndbg, as build by nix.
+        gdb_path = local_pwndbg_root / "result" / "bin" / "pwndbg"
+
+        if not gdb_path.exists():
+            print("ERROR: No nix-compatible pwndbg found. Run nix build .#pwndbg-dev")
+            sys.exit(1)
+    elif args.group == Group.CROSS_ARCH_USER:
+        # Some systems don't ship 'gdb-multiarch', but support multiple
+        # architectures in their regular binaries. Try the regular GDB.
+        supports_arches = "py import os; archs = ['i386', 'aarch64', 'arm', 'mips', 'riscv', 'sparc']; os._exit(3) if len([arch for arch in archs if arch in gdb.architecture_names()]) == len(archs) else os._exit(2)"
+
+        gdb_path_str = shutil.which("pwndbg")
+        if gdb_path_str is None:
+            print("ERROR: No 'pwndbg' executables in path")
+            sys.exit(1)
+
+        result = subprocess.run([gdb_path_str, "-nx", "-ex", supports_arches], capture_output=True)
+        # GDB supports cross architecture targets
+        if result.returncode == 3:
+            gdb_path = Path(gdb_path_str)
+        else:
+            print("ERROR: 'pwndbg' does not support cross architecture targets")
+            sys.exit(1)
+    else:
+        # Use the regular system GDB.
+        gdb_path_str = shutil.which("pwndbg")
+        if gdb_path_str is None:
+            print("ERROR: No 'gdb' executable in path")
+            sys.exit(1)
+        gdb_path = Path(gdb_path_str)
+
+    from host.gdb import GDBTestHost
+
+    return GDBTestHost(
+        local_pwndbg_root,
+        local_pwndbg_root / args.group.library(),
+        local_pwndbg_root / args.group.binary_dir(),
+        gdb_path,
+    )
+
+
+def get_lldb_host(args: argparse.Namespace, local_pwndbg_root: Path) -> TestHost:
+    """
+    Build an LLDB-based test host.
+    """
+    if args.nix:
+        # Use pwndbg, as built by nix.
+        lldbinit_dir = local_pwndbg_root / "result" / "share" / "pwndbg/"
+
+        if not lldbinit_dir.exists():
+            print("ERROR: No nix-compatible pwndbg found. Run nix build .#pwndbg-lldb")
```

Comment:
it should be `nix build .#pwndbg-lldb-dev` with `-dev` suffix?

Reply:
As far as this PR goes, `pwndbg-lldb` and `pwndbg-lldb-dev` work the same. But the next few PRs do change it to `pwndbg-lldb-dev` down the line.